plasmapy/atomic/tests/test_ionization_states.py

```diff
@@ -0,0 +1,304 @@
+                f"The elements whose IonizationStates are being kept "
+                f"track of ({elements}) are not a subset of the "
+                f"elements whose abundances are being kept track of "
+                f"({elements_from_abundances}) for test {test}."
+            )
+
+    @pytest.mark.parametrize('test', test_names2)
+    def test_element_sorting(self, test):
+        elements = self.instances[test].elements
+        before_sorting = []
+        for element in elements:
+            atomic_numb = atomic_number(element)
+            try:
+                mass_numb = mass_number(element)
+            except InvalidIsotopeError:
+                mass_numb = 0
+            before_sorting.append((atomic_numb, mass_numb))
+        after_sorting = sorted(before_sorting)
+
+        assert before_sorting == after_sorting, (
+            f"Elements/isotopes are not sorted for test='{test}':\n" 
+            f"  before_sorting = {before_sorting}\n"
+            f"   after_sorting = {after_sorting}\n"
+            f"where above is (atomic_number, mass_number if isotope else 0)")
+
+    @pytest.mark.parametrize('test', test_names2)
+    def test_ionic_fractions(self, test):
+
+        errmsg = ""
+
+        elements = self.instances[test].elements
+        input_keys = tests[test]["inputs"].keys()
+        for element, input_key in zip(elements, input_keys):
+
+            expected = np.array(tests[test]["inputs"][input_key])
+
+            if isinstance(expected, u.Quantity):
+                expected = np.array(expected.value / np.sum(expected.value))
+
+            #if not isinstance(expected, np.ndarray)
+
+            actual = self.instances[test].ionic_fractions[element]
+
+            if not np.allclose(actual, expected):
+                errmsg += (
+                    f"\n\nThere is a discrepancy in ionic fractions for "
+                    f"({test}, {element}, {input_key})\n"
+                    f"  expected = {expected}\n"
+                    f"    actual = {actual}")
+            if not isinstance(actual, np.ndarray) or isinstance(actual, u.Quantity):
+                raise AtomicError(
+                    f"\n\nNot a numpy.ndarray: ({test}, {element})")
+
+        if errmsg:
+            raise AtomicError(errmsg)
+
+#    @pytest.mark.parametrize('test', [test for test in tests if ])
+
+    @pytest.mark.parametrize('test', test_names2)
+    def test_getitem_element(self, test):
+        """Test that __get_item__ returns an IonizationState instance"""
+        instance = self.instances[test]
+
+        for key in instance.elements:
+
+            try:
+                expected = instance.ionic_fractions[key]
+            except Exception as exc:
+                raise AtomicError(
+                    f"Unable to get ionic_fractions for '{key}' in "
+                    f"test='{test}'.") from exc
+
+            try:
+                actual = instance[key].ionic_fractions
+            except Exception as exc:
```

Comment:
I don't think we need `as exc` if its not being used when re-throwing the error.